include/aspect/material_model/replace_lithosphere_viscosity.h

```diff
@@ -0,0 +1,148 @@
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef _aspect_material_model_replace_lithosphere_viscosity_h
+#define _aspect_material_model_replace_lithosphere_viscosity_h
+
+#include <aspect/material_model/interface.h>
+#include <aspect/simulator_access.h>
+#include <aspect/utilities.h>
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    using namespace dealii;
+
+    /**
+     * A material model that applies a given constant viscosity in the lithosphere.
+     * Viscosity below this is taken from a ''base mode''chosen from any of the
+     * other available material models. The replace lithosphere viscosity
+     * material model allows the user to specify the depth of the lithosphere-asthenosphre
+     * boundary either as one value or as a file. All other properties are derived
+     * from the base model.
+     * @ingroup MaterialModels
+     */
+    template <int dim>
+    class ReplaceLithosphereViscosity : public MaterialModel::Interface<dim>, public ::aspect::SimulatorAccess<dim>
+    {
+      public:
+        /**
+         * Empty Constructor.
+         */
+        ReplaceLithosphereViscosity ();
+
+        /**
+         * Initialize the base model at the beginning of the run.
+         */
+        virtual
+        void initialize();
+
+        /**
+         * Function to compute the material properties in @p out given the
+         * inputs in @p in.
+         */
+        virtual
+        void
+        evaluate (const typename Interface<dim>::MaterialModelInputs &in,
+                  typename Interface<dim>::MaterialModelOutputs &out) const;
+        /**
+         * Method to declare parameters related to depth-dependent model
+         */
+        static void
+        declare_parameters (ParameterHandler &prm);
+
+        /**
+         * Method to parse parameters related to depth-dependent model
+         */
+        virtual void
+        parse_parameters (ParameterHandler &prm);
+
+        /**
+         * Method that indicates whether material is compressible. Depth dependent model is compressible
+         * if and only if base model is compressible.
+         */
+        virtual bool is_compressible () const;
+
+        /**
+         * Method to calculate reference viscosity for the model. The reference
+         * viscosity is simply the reference  viscosity from the base model.
+         */
+        virtual double reference_viscosity () const;
+
+      private:
+        /**
+         * Reads in file containing input data in ascii format.
+         */
+        Utilities::AsciiDataLookup<2> lab_depths;
```

Comment:
This feels like a lot of duplication with the `LithosphereMask` initial temperature plugin. We can avoid that if you would create a new helper class in `include/aspect/initial_temperature.h` that looks like this:
```
namespace LABDepth
{
  class LABDepthLookup: public SimulatorAccess<dim>
    public:
        /**
         * Return LAB depth as a function of position (latitude and longitude). For the
         * current class, this function returns value from the text file. We read in
         * two dimensions so the third column (depth) is treated as data.
         */
        double
        ascii_lab (const Point<2> &position) const;

        /**
         * Declare the parameters this class takes through input files.
         */
        static
        void
        declare_parameters (ParameterHandler &prm);

        /**
         * Read the parameters this class declares from the parameter file.
         */
        void
        parse_parameters (ParameterHandler &prm);

private:
     all of the variables the class needs
}
```
You can then use this class from the initial temperature plugin *and* from the material model, and the two implementation always stay the same (you will need to include the initial temperature header here in the material model). Plus if somebody adds a new way to define the LAB both plugins immediately get access. 
Splt this up in two pull requests. The first only creates the class for initial temperature as I wrote above. The second is this one, which can then just use the new class.